functions/callback-functions.md

```diff
@@ -0,0 +1,214 @@
+
+The variable `bakeCake` references a different function that prints details about baking a cake.
+
+When we bake a cake, maybe there's always cleanup behavior that we want to call, whether it's cleaning our counters, sink, or stove-top.
+
+We can _change the function signature_ of `bakeCake` to accept an argument, `cleanupBehavior`. Inside our `bakeCake` function, we can take that argument and _invoke_ whatever function it references, by attaching `()` to it.
+
+<!-- prettier-ignore-start -->
+```js
+const bakeCake = function (cleanupBehavior) {
+  console.log('Make the batter...');
+  console.log('Put it in the oven...');
+  cleanupBehavior();
+}
+```
+<!-- prettier-ignore-end -->
+
+We can refer to `cleanupBehavior` as a _callback function_ inside our `bakeCake` function. After we look at the big picture, we can think about the phrase "callback."
+
+<!-- prettier-ignore-start -->
+```js
+const cleanCountertop = function () {
+  console.log('I\'m cleanin\' my counters now! Scrubba scrub scrub.');
+}
+
+const bakeCake = function (cleanupBehavior) {
+  console.log('Make the batter...');
+  console.log('Put it in the oven...');
+  cleanupBehavior();
+}
+
+bakeCake(cleanCountertop);
+```
+<!-- prettier-ignore-end -->
+
+When we ultimately _invoke_ the `bakeCake` function with `bakeCake()`, we need to pass in an argument.
+
+<!-- prettier-ignore-start -->
+```js
+bakeCake(cleanCountertop);
+```
+<!-- prettier-ignore-end -->
+
+We're passing in `cleanCountertop`, the variable that refers to our cleaning function. We're specifically _not_ invoking the function. If we were _invoking_ it, we would use `cleanCountertop()`.
+
+Reviewing our `bakeCake` function, we can see that we invoke our `cleanupBehavior` function with `cleanupBehavior()`, without knowing if it's cleaning the countertop, cleaning the sink, cleaning the stove-top, or cleaning the bathroom!
+
+<!-- prettier-ignore-start -->
+```js
+const bakeCake = function (cleanupBehavior) {
+  console.log('Make the batter...');
+  console.log('Put it in the oven...');
+  cleanupBehavior();
+}
+```
+<!-- prettier-ignore-end -->
+
+We call `cleanupBehavior` a _callback function_ inside `bakeCake`. It refers to the expectation that `bakeCake` should _call back_ a function... the function that was passed in, when `bakeCake` was invoked in the first place.
+
+## Revisiting Calculator
+
+Let's consider how we can build our calculator function using callbacks. This process will be very similar to our `bakeCake` example, but we'll be mindful about how arguments are passed in.
+
+Consider this code:
+
+<!-- prettier-ignore-start -->
+```js
+const multiply = function (factor_a, factor_b) {
+  console.log('Now we\'re in multiply~~');
+  return factor_a * factor_b;
+}
+
+const calculate = function (a, b, operate) {
+  console.log('We\'re inside the calculate function!');
+  return operate(a, b);
+}
+
+const result = calculate(7, 6, multiply);
+console.log('Our final result from calculate is:', result);
```

Comment:
call with both multiply and add
rename the const vars as a result of two calls
```suggestion
const multiply_result = calculate(7, 6, multiply);
console.log('Our final result from calculate is:', multiply_result);

const add_result = calculate(7, 6, add);
console.log('Our final result from calculate is:', add_result);
```

Reply:
We could consider doing this in a loop, which I find a little more compelling and would challenge the reader to do some research about for/of. But it would require some rework to the following passages.

```js
for (const operator of [multiply, add]) {
  const result = calculate(7, 6, operator);
  console.log('Our final result from calculate is:', result);
}
```